cmd/modelcmd/confirmation.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelcmd
+
+import (
+	"fmt"
+	"github.com/juju/cmd/v3"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	jujucmd "github.com/juju/juju/cmd"
+)
+
+// ConfirmationCommandBase provides common attributes and methods that
+// commands require to confirm the execution.
+type ConfirmationCommandBase struct {
+	assumeYes      bool // DEPRECATED
+	assumeNoPrompt bool
+}
+
+// SetFlags implements Command.SetFlags.
+func (c *ConfirmationCommandBase) SetFlags(f *gnuflag.FlagSet) {
+	f.BoolVar(&c.assumeYes, "y", false, "Do not ask for confirmation. Option present for backwards compatibility with Juju 2.9")
+	f.BoolVar(&c.assumeYes, "yes", false, "")
+	f.BoolVar(&c.assumeNoPrompt, "no-prompt", false, "Do not ask for confirmation")
+}
+
+// Init implements Command.Init.
+func (c *ConfirmationCommandBase) Init(args []string) error {
+	if !c.assumeNoPrompt {
+		assumeNoPrompt, skipErr := jujucmd.CheckSkipConfirmEnvVar()
+		if skipErr != nil && !errors.Is(skipErr, errors.NotFound) {
+			return errors.Trace(skipErr)
+		}
+		if !errors.Is(skipErr, errors.NotFound) {
+			c.assumeNoPrompt = assumeNoPrompt
+		}
+	}
+	return nil
+}
+
+// Run implements Command.Run
+func (c *ConfirmationCommandBase) Run(ctx *cmd.Context) error {
+	if c.assumeYes {
+		fmt.Fprint(ctx.Stdout, "WARNING: '-y'/'--yes' flags are deprecated and will be removed in Juju 3.1\n")
+	}
+	return nil
+}
+
+// GetAssumeYes returns assumeYes
+func (c *ConfirmationCommandBase) GetAssumeYes() bool {
+	return c.assumeYes
+}
+
+// GetAssumeNoPrompt returns assumeNoPrompt
+func (c *ConfirmationCommandBase) GetAssumeNoPrompt() bool {
+	return c.assumeNoPrompt
+}
```

Comment:
I think these two functions can be made a little smarter. Every time these function are called they are combined with an `or`, and I can't think of any use case of having them separate (outside of this struct itself).

Getter functions which directly return private attributes can sometimes be a bit of a code smell, indicating you may be revealing implementation details or delegating responsibility to users which should really be kept here. When we drop the -y flag, users of this struct shouldn't notice

So I would drop these and replace with `func (c *ConfirmationCommandBase) NeedsConfirmation() bool {}` or something which is just passed directly into if statements above

Reply:
Agreed, thx